uc-crux-llvm/src/UCCrux/LLVM/Run/Simulate.hs

```diff
@@ -555,6 +565,16 @@ mkCallbacks appCtx modCtx funCtx halloc callbacks constraints cfg llvmOpts =
          argShapes <- fromMaybe rd <$> IORef.readIORef argShapeRef
          resHook sym mem argShapes cruxResult ucCruxResult
 
+    traces :: Crux.CruxSimulationResult -> Set [What4.ProgramLoc]
+    traces =
+      let go =
+            \case
+              Crux.Branch pgs1 pgs2 -> go pgs1 ++ go pgs2
```

Comment:
This loop is very close to `crux`'s [`provedGoalLocs`](https://github.com/GaloisInc/crucible/blob/a6669a5fe657edecc61e294dd706e04ebb071776/crux/src/Crux/Report.hs#L84-L90) function. Is it possible to just use that instead? Note that `provedGoalLocs` returns more `ProgramLoc`s than this code does, as `provedGoalLocs` also includes `simErrorLoc`s—I'm not sure if that is a deal-breaker for `uc-crux-llvm`.

Reply:
Ah, thanks for pointing out that function!

Reply:
I generalized the function in Crux, let me know what you think.